### PR TITLE
SDK Release for desktopvirtualization based on 2022-09-09 api (latest stable)

### PR DESCRIPTION
**Pull request by alec-baird**
…ice with tag package-2022-09 (latest stable)

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 1]
```diff
@@ -63,7 +65,6 @@ public partial class HostPoolData : Azure.ResourceManager.Models.TrackedResource
         public Azure.ResourceManager.DesktopVirtualization.Models.HostPoolLoadBalancerType LoadBalancerType { get { throw null; } set { } }
         public Azure.Core.ResourceIdentifier ManagedBy { get { throw null; } set { } }
         public int? MaxSessionLimit { get { throw null; } set { } }
-        public Azure.ResourceManager.DesktopVirtualization.Models.DesktopVirtualizationMigrationProperties MigrationRequest { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:66]: This is a breaking change that now allow, please use [custom code](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cognitiveservices/Azure.ResourceManager.CognitiveServices/src/Customized/Models/ServiceAccountEncryptionProperties.cs) to add this property back and mark it as `Obsolete` and `EditorBrowsableState.Never`

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 2]
```diff
@@ -164,22 +165,22 @@ public partial class ScalingPlanCollection : Azure.ResourceManager.ArmCollection
...
         System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.DesktopVirtualization.ScalingPlanResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.DesktopVirtualization.ScalingPlanResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
         System.Collections.Generic.IEnumerator<Azure.ResourceManager.DesktopVirtualization.ScalingPlanResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.DesktopVirtualization.ScalingPlanResource>.GetEnumerator() { throw null; }
         System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
     }
     public partial class ScalingPlanData : Azure.ResourceManager.Models.TrackedResourceData
     {
-        public ScalingPlanData(Azure.Core.AzureLocation location) : base (default(Azure.Core.AzureLocation)) { }
+        public ScalingPlanData(Azure.Core.AzureLocation location, string timeZone) : base (default(Azure.Core.AzureLocation)) { }
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:177]: This is a breaking change, which should not change an optional property to required, please fix it from swagger spec

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 3]
```diff
@@ -164,22 +165,22 @@ public partial class ScalingPlanCollection : Azure.ResourceManager.ArmCollection
...
         public string Description { get { throw null; } set { } }
         public Azure.ETag? ETag { get { throw null; } }
         public string ExclusionTag { get { throw null; } set { } }
         public string FriendlyName { get { throw null; } set { } }
         public System.Collections.Generic.IList<Azure.ResourceManager.DesktopVirtualization.Models.ScalingHostPoolReference> HostPoolReferences { get { throw null; } }
-        public Azure.ResourceManager.DesktopVirtualization.Models.HostPoolType? HostPoolType { get { throw null; } set { } }
+        public Azure.ResourceManager.DesktopVirtualization.Models.ScalingHostPoolType? HostPoolType { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:183]: This is a breaking change, can still use the old type?

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 4]
```diff
@@ -493,11 +547,31 @@ public partial class VirtualWorkspaceResource : Azure.ResourceManager.ArmResourc
 }
 namespace Azure.ResourceManager.DesktopVirtualization.Models
 {
-    public partial class DesktopVirtualizationMigrationProperties
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:496]: Please add this back to keep backward compatible.

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 5]
```diff
@@ -493,11 +547,31 @@ public partial class VirtualWorkspaceResource : Azure.ResourceManager.ArmResourc
...
+    }
+    public partial class AgentUpdateProperties
+    {
+        public AgentUpdateProperties() { }
+        public System.Collections.Generic.IList<Azure.ResourceManager.DesktopVirtualization.Models.MaintenanceWindowProperties> MaintenanceWindows { get { throw null; } }
+        public string MaintenanceWindowTimeZone { get { throw null; } set { } }
+        public Azure.ResourceManager.DesktopVirtualization.Models.SessionHostComponentUpdateType? SessionHostComponentUpdateType { get { throw null; } set { } }
+        public bool? UseSessionHostLocalTime { get { throw null; } set { } }
+    }
+    public enum DayOfWeek
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:566]: rename this to `DesktopVirtualizationDayOfWeek` by using `rename-mapping`

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 6]
```diff
@@ -493,11 +547,31 @@ public partial class VirtualWorkspaceResource : Azure.ResourceManager.ArmResourc
 }
 namespace Azure.ResourceManager.DesktopVirtualization.Models
 {
-    public partial class DesktopVirtualizationMigrationProperties
+    public partial class AgentUpdatePatchProperties
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:550]: `AgentUpdate` is a little bit generic, can you add a prefix?

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 7]
```diff
@@ -670,26 +745,17 @@ public partial class HostPoolRegistrationInfoPatch
         public static bool operator !=(Azure.ResourceManager.DesktopVirtualization.Models.HostPoolType left, Azure.ResourceManager.DesktopVirtualization.Models.HostPoolType right) { throw null; }
         public override string ToString() { throw null; }
     }
-    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
-    public readonly partial struct MigrationOperation : System.IEquatable<Azure.ResourceManager.DesktopVirtualization.Models.MigrationOperation>
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:674]: Please add this back by custom code to keep backward compatible.

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 8]
```diff
@@ -782,39 +848,76 @@ public partial class ScalingHostPoolReference
...
+        public override string ToString() { throw null; }
+    }
     public partial class ScalingPlanPatch
     {
         public ScalingPlanPatch() { }
         public string Description { get { throw null; } set { } }
         public string ExclusionTag { get { throw null; } set { } }
         public string FriendlyName { get { throw null; } set { } }
         public System.Collections.Generic.IList<Azure.ResourceManager.DesktopVirtualization.Models.ScalingHostPoolReference> HostPoolReferences { get { throw null; } }
-        public Azure.ResourceManager.DesktopVirtualization.Models.HostPoolType? HostPoolType { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:792]: This is a breaking change that is not allowed, please use [custom code](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cognitiveservices/Azure.ResourceManager.CognitiveServices/src/Customized/Models/ServiceAccountEncryptionProperties.cs) to add this property back and mark it as Obsolete and EditorBrowsableState.Never

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs [hunk 9]
```diff
@@ -782,39 +848,76 @@ public partial class ScalingHostPoolReference
...
+        public int? RampUpMinimumHostsPct { get { throw null; } set { } }
+        public Azure.ResourceManager.DesktopVirtualization.Models.Time RampUpStartTime { get { throw null; } set { } }
+    }
     public partial class ScalingSchedule
     {
         public ScalingSchedule() { }
         public System.Collections.Generic.IList<Azure.ResourceManager.DesktopVirtualization.Models.ScalingScheduleDaysOfWeekItem> DaysOfWeek { get { throw null; } }
         public string Name { get { throw null; } set { } }
         public Azure.ResourceManager.DesktopVirtualization.Models.SessionHostLoadBalancingAlgorithm? OffPeakLoadBalancingAlgorithm { get { throw null; } set { } }
-        public System.DateTimeOffset? OffPeakStartOn { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:803]: This is a breaking change that is not allowed, please use [custom code](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cognitiveservices/Azure.ResourceManager.CognitiveServices/src/Customized/Models/ServiceAccountEncryptionProperties.cs) to add this property back and mark it as Obsolete and EditorBrowsableState.Never

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs (hunk 1 above)
Comment (alec-baird) [reply sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:66]: Do I need to make any changes to swagger? Or can I just change the code here in the .netstandard2.0.cs file?

**Comment by alec-baird** [comment]
@JeffreyRichter , as you approved our stable with the breaking changes, can you provide context on how we can get the SDK released with the breaking changes?

**Comment by JeffreyRichter** [comment]
> @JeffreyRichter , as you approved our stable with the breaking changes, can you provide context on how we can get the SDK released with the breaking changes?

Do you have a link to the swagger PR where the breaking changes were made?

**Comment by alec-baird** [comment]
> > @JeffreyRichter , as you approved our stable with the breaking changes, can you provide context on how we can get the SDK released with the breaking changes?
> 
> Do you have a link to the swagger PR where the breaking changes were made?

https://github.com/Azure/azure-rest-api-specs/pull/20683

**Comment by JeffreyRichter** [comment]
@ArthurMa1978, The Azure Breaking Change review board approved the swagger breaks that are causing these SDK breaks (I think). Since this is a control plane SDK, that is code gen'd, the back compat issues can't be fixed unless the .NET SDK team expects these to be manually fixed. What is the .NET SDK team's expectation here?

File: sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:66]: Swagger change is the better way

Comment (ArthurMa1978) [reply sdk/desktopvirtualization/Azure.ResourceManager.DesktopVirtualization/api/Azure.ResourceManager.DesktopVirtualization.netstandard2.0.cs:66]: Ok, let's use the custom code to fix all breaking chagnes

**Comment by alec-baird** [comment]
@ArthurMa1978 , it has been 2 weeks since we have last heard anything on this PR. Can you take a look?

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**